Classes/Filter/EventStreamFilter.php

```diff
@@ -0,0 +1,235 @@
+<?php
+namespace Neos\EventStore\Filter;
+
+/*
+ * This file is part of the Neos.EventStore package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\Flow\Annotations as Flow;
+
+/**
+ * EventStreamFilter
+ */
+class EventStreamFilter
+{
+    /**
+     * @var int
+     */
+    public $minimumVersion = null;
+
+    /**
+     * @var int
+     */
+    public $maximumVersion = null;
+
+    /**
+     * @var \DateTimeImmutable
+     */
+    public $since = null;
+
+    /**
+     * @var \DateTimeImmutable
+     */
+    public $until = null;
```

Comment:
Can we omit `since` and `until` for now or at least put it in a specialized filter that is not part of the main API. We won't be able to implement time based filters with the GES adapter for example